test/SymbolGtest.cpp

```diff
@@ -0,0 +1,184 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   SymbolGtest.cpp
+ *  @date   Wed February 23 2022
+ *  @brief  Unit test suite for the Sample class
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#include <limits>
+#include <sstream>
+
+#include<gtest/gtest.h>
+#include<gmock/gmock.h>
+#include "../inc/Symbol.h"
+
+using namespace testing;
+
+class GivenInitializedSymbolsSomeWithValues: public ::testing::Test
+{
+    protected:
+        static constexpr int FIRST_INT_VALUE = 42;
+        static constexpr int SECOND_INT_VALUE = 43;
+        static constexpr int THIRD_INT_VALUE = 41;
+        static constexpr int FOURTH_INT_VALUE = 44;
+        static constexpr double FIRST_DOUBLE_VALUE = 42.42e10;
+        static constexpr double SECOND_DOUBLE_VALUE = 43.43e10;
+        static constexpr double THIRD_DOUBLE_VALUE = 41.41e10;
+        static constexpr double FOURTH_DOUBLE_VALUE = 44.44e10;
+        static constexpr const char *BAD_USER_INPUT = "asdf";
+
+        calculator::Symbol<int> intSymbolA;
+        calculator::Symbol<int> intSymbolBBound;
+        calculator::Symbol<int> intSymbolDBound;
+        calculator::Symbol<double> doubleSymbolA;
+        calculator::Symbol<double> doubleSymbolDBound;
+        calculator::Symbol<double> doubleSymbolBBound;
+
+        // Using SetUp() is normally preferred to having a ctor.
+        // In this case, we can't get away from initializing in the ctor since 
+        // the instance variables require symbols to be set on construction
+        GivenInitializedSymbolsSomeWithValues(): 
+                intSymbolA{'a'},
+                intSymbolBBound{'B'},
+                intSymbolDBound{'d'},
+                doubleSymbolA{'A'},
+                doubleSymbolBBound{'B'},
+                doubleSymbolDBound{'d'}
+        {
+        }
+
+        void SetUp() override
+        {
+            intSymbolBBound.setValue(FIRST_INT_VALUE);
+            intSymbolDBound.setValue(FIRST_INT_VALUE);
+            doubleSymbolBBound.setValue(FIRST_DOUBLE_VALUE);
+            doubleSymbolDBound.setValue(FIRST_DOUBLE_VALUE);
+        }
+};
+constexpr int GivenInitializedSymbolsSomeWithValues::FIRST_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::SECOND_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::THIRD_INT_VALUE;
+constexpr int GivenInitializedSymbolsSomeWithValues::FOURTH_INT_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::FIRST_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::SECOND_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::THIRD_DOUBLE_VALUE;
+constexpr double GivenInitializedSymbolsSomeWithValues::FOURTH_DOUBLE_VALUE;
+constexpr const char *GivenInitializedSymbolsSomeWithValues::BAD_USER_INPUT;
+
+
+
+TEST_F(GivenInitializedSymbolsSomeWithValues,
```

Comment:
Unit  test also need doxygen style comments

Reply:
Hey Max, could you give me an example of how a doxygen comment should look for a test case?